test/parallel/test-http-client-spurious-aborted.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const common = require('../common');
+const http = require('http');
+const assert = require('assert');
+const fs = require('fs');
+const Countdown = require('../common/countdown');
+
+function cleanup(fname) {
+  try {
+    if (fs.statSync(fname)) fs.unlinkSync(fname);
+  } catch (err) {}
+}
+
+const N = 2;
+const fname = '/dev/null';
+let abortRequest = true;
+
+const server = http.Server(common.mustCall((req, res) => {
+  const headers = { 'Content-Type': 'text/plain' };
+  headers['Content-Length'] = 50;
+  const socket = res.socket;
+  res.writeHead(200, headers);
+  setTimeout(() => res.write('aaaaaaaaaa'), 100);
+  setTimeout(() => res.write('bbbbbbbbbb'), 200);
+  setTimeout(() => res.write('cccccccccc'), 300);
+  setTimeout(() => res.write('dddddddddd'), 400);
+  if (abortRequest) {
+    setTimeout(() => socket.destroy(), 600);
+  } else {
+    setTimeout(() => res.end('eeeeeeeeee'), 1000);
+  }
+}, N));
+
+server.listen(0, common.mustCall(() => {
+  cleanup(fname);
+  download();
+}));
+
+const finishCountdown = new Countdown(N, common.mustCall(() => {
+  server.close();
+}));
+const reqCountdown = new Countdown(N, common.mustCall());
```

Comment:
I would prefer if `N` was not a constant, but rather it uses a two steps approach, maybe with some helpers.

It's not clear why you are calling twice too.

Reply:
it's running two tests - one to ensure we don't abort a good request, one to ensure we do abort a bad one. i can refactor to take out the countdown timers altogether. was trying to stay close to how other tests are constructed. if you can suggest how to change i am happy to refactor.